Sources/SwiftFusion/Optimizers/CGLS.swift

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/// Conjugate Gradient Least Squares (CGLS) optimizer.
+///
+/// An optimizer that implements CGLS second order optimizer
+public class CGLS {
+  /// The set of steps taken.
+  public var step: Int = 0
+
+  /// Constructor (Probably not necessary)
+  public init() {}
+
+  /// Optimize the Gaussian Factor Graph with a initial estimate
+  /// Reference: Bjorck96book_numerical-methods-for-least-squares-problems
+  /// Page 289, Algorithm 7.4.1
+  public func optimize(gfg: GaussianFactorGraph, initial: Values) {
+    step += 1
+    
+    let b = gfg.b
+    let A = gfg.A
```

Comment:
I don't think that makes sense. We should use the vjp and jvp operators of the factor graph.